src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -327,6 +331,36 @@ public static void deleteOnDiskFile(long windowStartTime) {
     }
   }
 
+  /**
+   * Returns the timestamps associated with on-disk files.
+   *
+   * @return the timestamps associated with on-disk files
+   */
+  public static Set<Long> listOnDiskFiles() {
+    String prefix = PluginSettings.instance().getSettingValue(DB_FILE_PREFIX_PATH_CONF_NAME, DB_FILE_PREFIX_PATH_DEFAULT);
+    Path prefixPath = Paths.get(prefix);
+    Path parentPath = prefixPath.getParent();
+    Set<Long> found = new HashSet<Long>();
+    try (Stream<Path> paths = Files.list(parentPath)) {
+      PathMatcher matcher = FileSystems.getDefault().getPathMatcher("regex:" + prefix + "\\d+");
```

Comment:
We might skip the overhead of regex compiling by stripping off the `parentPath` from the `prefix` -  the remainder is what we are looking for in the directory.